client/server/pages/index.js

```diff
@@ -103,7 +103,16 @@ function getDefaultContext( request, response, entrypoint = 'entry-main' ) {
 	}
 
 	const cacheKey = `${ getNormalizedPath( request.path, request.query ) }:gdpr=${ showGdprBanner }`;
-	const cachedServerState = stateCache.get( cacheKey ) || {};
+
+	/**
+	 * A cache object can be written for an SSR route like /themes when a request
+	 * is logged out. To avoid using that logged-out data for an authenticated
+	 * request, we should not utilize the state cache for logged-in requests.
+	 * Note that in dev mode (when the user is not bootstrapped), all requests
+	 * are considered logged out. This shouldn't cause issues because only one
+	 * user is using the cache in dev mode -- so cross-pollination won't happen.
```

Comment:
Nitpick: I think we're worried about cross-(request) _pollution_ rather than cross-_pollination_ 🐝 

```suggestion
	 * user is using the cache in dev mode -- so cross-request pollution won't happen.
```

Reply:
Fair enough! I think I was attempting to get at the same idea, but I guess "cross-pollination" isn't a phrase people use 😆